src/result.rs

```diff
@@ -181,6 +183,14 @@ impl CqlValueWrapper {
     }
   }
 
+  #[napi]
+  pub fn get_duration(&self) -> napi::Result<DurationWrapper> {
+    match self.internal.as_cql_duration() {
+      Some(r) => Ok(DurationWrapper::from_cql_duration(r)),
+      None => Err(Error::new(Status::GenericFailure, "Error")),
+    }
+  }
```

Comment:
Of course the error message must be more informative. Maybe `Error: the underlying CQL value is not of Duration type`?

Reply:
Also, please check which `Status` variant is the best suit here. Perhaps it's `GenericFailure` indeed, but let's make sure.

Reply:
Here is list of all possible ``Status`` values: [code](https://github.com/napi-rs/napi-rs/blob/de3a60dfc1e59b3a0a493cf8d955506f7ab39d86/crates/napi/src/status.rs#L7-L35) I can't find any appropriate documentation, but it seems all other types are for receiving invalid type of argument.